types/react-native/index.d.ts

```diff
@@ -469,6 +471,84 @@ export interface Insets {
     right?: number;
 }
 
+export interface PressableStateCallbackType {
+    pressed: boolean;
+}
+
+export interface PressableProps extends AccessibilityProps, Omit<ViewProps, 'style' | 'hitSlop'> {
+    /**
+     * Called when a single tap gesture is detected.
+     */
+    onPress?: (event: GestureResponderEvent) => void;
+
+    /**
+     * Called when a touch is engaged before `onPress`.
+     */
+    onPressIn?: (event: GestureResponderEvent) => void;
+
+    /**
+     * Called when a touch is released before `onPress`.
+     */
+    onPressOut?: (event: GestureResponderEvent) => void;
+
+    /**
+     * Called when a long-tap gesture is detected.
+     */
+    onLongPress?: (event: GestureResponderEvent) => void;
+
+    /**
+     * Either children or a render prop that receives a boolean reflecting whether
+     * the component is currently pressed.
+     */
+    children: React.ReactNode | ((state: PressableStateCallbackType) => React.ReactNode);
+
+    /**
+     * Duration (in milliseconds) from `onPressIn` before `onLongPress` is called.
+     */
+    delayLongPress?: number;
+
+    /**
+     * Whether the press behavior is disabled.
+     */
+    disabled?: boolean;
+
+    /**
+     * Additional distance outside of this view in which a press is detected.
+     */
+    hitSlop?: Insets | number;
+
+    /**
+     * Additional distance outside of this view in which a touch is considered a
+     * press before `onPressOut` is triggered.
+     */
+    pressRetentionOffset?: Insets | number;
+
+    /**
+     * If true, doesn't play system sound on touch.
+     */
+    android_disableSound?: boolean;
+
+    /**
+     * Enables the Android ripple effect and configures its color.
+     */
+    android_ripple?: RippleBackgroundPropType;
+
+    /**
+     * Used only for documentation or testing (e.g. snapshot testing).
+     */
+    testOnly_pressed?: boolean;
+
+    /**
+     * Either view styles or a function that receives a boolean reflecting whether
+     * the component is currently pressed and returns view styles.
+     */
+    style?: StyleProp<ViewStyle> | ((state: PressableStateCallbackType) => StyleProp<ViewStyle>);
+}
+
+declare class PressableComponent extends React.Component<PressableProps> {}
+declare const PressableBase: Constructor<NativeMethodsMixinType> & typeof PressableComponent;
+export class Pressable extends PressableBase {}
```

Comment:
Hmm, the upstream type is an ‘abstract component’ (for which we don’t have a equivalent yet), but I don’t believe it needs the native methods, no? (The component itself is also a function based one, not a class based one.)

Reply:
@alexbrazier This is my only remaining question—doesn’t seem like a blocking thing, though.

Reply:
Ah, missed this one - any thoughts on what this would look like?